scripts/utils/fuels/relayCommonMessage.ts

```diff
@@ -0,0 +1,140 @@
+/// @dev The Fuel testing utils.
+/// A set of useful helper methods for the integration test environment.
+import {
+  Message,
+  WalletUnlocked as FuelWallet,
+  ZeroBytes32,
+  ScriptTransactionRequest,
+  TransactionRequestLike,
+  arrayify,
+  InputType,
+  hexlify,
+  OutputType,
+  TransactionResponse,
+  Predicate,
+} from 'fuels';
+import { readFileSync } from 'fs';
+import { join } from 'path';
+import { debug } from '../logs';
+
+const brigdePredicateBytecode = readFileSync(
+  join(__dirname, '../../../bridge-message-predicates/contract_message_predicate.bin')
+);
+const brigdeScriptBytecode = readFileSync(
+  join(__dirname, '../../../bridge-message-predicates/contract_message_script.bin')
+);
+
+// Create a predicate contract for common messages
+const predicate = new Predicate(hexlify(brigdePredicateBytecode), 0);
+
+// Details for relaying common messages with certain predicate roots
+const COMMON_RELAYABLE_MESSAGES: CommonMessageDetails[] = [
+  {
+    name: 'Message To Contract v1.3',
+    predicateRoot: predicate.address.toHexString(),
+    predicate: hexlify(brigdePredicateBytecode),
+    script: hexlify(brigdeScriptBytecode),
+    buildTx: async (
+      relayer: FuelWallet,
+      message: Message,
+      details: CommonMessageDetails,
+      txParams: Pick<TransactionRequestLike, 'gasLimit' | 'gasPrice' | 'maturity'>
+    ): Promise<ScriptTransactionRequest> => {
+      //TODO: minGas should be much lower and more in line with what the predicate actually verifies (currently 1200000)
+      const minGas: number = 500000000000;
+      const script = arrayify(details.script);
+      const predicate = arrayify(details.predicate);
+
+      // find a UTXO that can cover gas costs
+      let coins = (await relayer.getCoins()).filter((coin) => coin.assetId == ZeroBytes32 && coin.amount.gt(minGas));
+      if (coins.length == 0) throw new Error('wallet has no single UTXO that can cover gas costs');
```

Comment:
Can we not sum up UTXOs to cover cost or must it be 1 UTXO?

Reply:
This code was just copied from the old code base. I updated it to use to get resources to spend, making it to work with multiple UTXOs.

Reply:
Comparing the previous code to the current change I think this might result in a bug. The previous code searched for a single coin that can cover the minimum gas cost but the latest change filters for coins without making any guarantee that the sum of the coins is greater than the minimum cost. Do we need to add another check here to make sure that the cost for gas is covered? It seems like that restriction is now gone

Reply:
The `getResoucesToSpend` already does the check if you don't have resources. The query fails and throws an error.

Reply:
My understanding is that it checks to see if it's a coin but it does not check if the coin has enough to cover the gas so we can have coins but the total sum of resources is less than the gas required and the Tx will fail.
Am I misunderstanding?